### PR TITLE
DOCSP-3828 - Add Set Parameter configuration file options.

### PR DESCRIPTION
**Pull request by jdestefano-mongo**
JIRA: https://jira.mongodb.org/browse/DOCSP-3832

Staged:
- Configuration file setting: https://docs-mongodbcom-staging.corp.mongodb.com/bi-connector/jdestefano/DOCSP-3828/reference/mongosqld.html#set-parameter-options
- Command-line option: https://docs-mongodbcom-staging.corp.mongodb.com/bi-connector/jdestefano/DOCSP-3828/reference/mongosqld.html#set-parameter-option

### CODE REVIEW

File: source/includes/options-mongosqld.yaml [hunk 1]
```diff
@@ -874,4 +874,22 @@ description: |
...
+args: <polymorphic_type_conversion_mode | type_conversion_mode>
+description: |
+  .. list-table::
+   :header-rows: 1
+   * - Name
+     - Type
+     - Corresponds to
+   * - ``polymorphic_type_conversion_mode``
+     - string
+     - Determines how :binary:`~bin.mongosqld` evaluates document fields that are specified with multiple data types.
```
Comment (atsansone) [inline source/includes/options-mongosqld.yaml]:
```suggestion
     - Determines how :binary:`~bin.mongosqld` evaluates document
       fields that are specified with multiple data types.
```
(Text wrap at 72 chars.)

File: source/includes/options-mongosqld.yaml [hunk 2]
```diff
@@ -874,4 +874,22 @@ description: |
...
+   * - ``type_conversion_mode``
+     - string
+     - Specifies the semantics that :binary:`~bin.mongosqld` uses for type conversions, such as the CAST function.
```
Comment (atsansone) [inline source/includes/options-mongosqld.yaml]:
```suggestion
     - Specifies which mode the :binary:`~bin.mongosqld` uses to
       :doc:`convert data types </reference/type-conversion>`.
       |bi-short| uses its own mode (``mongosql``) by default. Accepted
       values are ``mongosql`` and ``mysql``.
```
Wrap at 72 chars. Added link to data type conversion.

File: source/reference/mongosqld.txt [hunk 3]
```diff
@@ -680,6 +680,40 @@ Process Management Options
...
+       that are specified with multiple data types. For a list of
+       accepted values, see
+       :ref:`System Variables <system-environment-variables>`.
+
+   * - .. setting:: setParameter.type_conversion_mode
+     - string
+     - Specifies the semantics that :binary:`~bin.mongosqld` uses for
+       type conversions, such as the CAST function. For a list of
+       accepted values, see
+       :ref:`System Variables <system-environment-variables>`.
```
Comment (atsansone) [inline source/reference/mongosqld.txt]: I would replace this with the block I changed earlier in this PR:
```
     - Specifies which mode the :binary:`~bin.mongosqld` uses to
       :doc:`convert data types </reference/type-conversion>`.
       |bi-short| uses its own mode (``mongosql``) by default. Accepted
       values are ``mongosql`` and ``mysql``.
```
There were only two values, so just listing them here made sense.

File: source/reference/mongosqld.txt [hunk 4]
```diff
@@ -680,6 +680,40 @@ Process Management Options
...
+   :header-rows: 1
+
+   * - Name
+     - Type
+     - Corresponds to
+
+   * - .. setting:: setParameter.polymorphic_type_conversion_mode
+     - string
+     - Determines how :binary:`~bin.mongosqld` evaluates document fields
+       that are specified with multiple data types. For a list of
```
Comment (atsansone) [inline source/reference/mongosqld.txt]:
```suggestion
       that are specified with multiple data types. Accepted values are
       ``off``, ``fast``, and ``safe``. To learn more about these
       values,
```

File: source/reference/mongosqld.txt [hunk 5]
```diff
@@ -680,6 +680,40 @@ Process Management Options
...
+       accepted values, see
```
Comment (atsansone) [inline source/reference/mongosqld.txt]:
```suggestion
       see
```

**Comment by jdestefano-mongo** [comment]
@rychipman Mind taking a look at this docs PR when you have some bandwidth?

**Pull request merged**